src/shop.js

```diff
@@ -1,5 +1,6 @@
 function BinartaShopjs(checkpoint, deps) {
     var shop = this;
+    var application = deps.application || {profile: function () {}};
```

Comment:
Your lower code will still crash because profile will be undefined in the case of the fallback so your test should be for undefined on top of availablePaymentMethods.